src/ProductView/ProductView.js

```diff
@@ -103,6 +103,13 @@ class ProductView extends Component {
 
     onChangeFilterInput(currentFilterId, value) {
         const updatedFilters = this.state.filter;
+
+        if (!value) {
+            delete updatedFilters[currentFilterId];
```

Comment:
Have to revise my comment above. Due to the fact that updateFilters is an array instead of an object you should not use "delte" here. Delete only removes the reference and writes undefined to the corresponding index. This makes the array "holey". In general please use splice for removing elements of an array. In this case it becomes complicated to use splice, because the implementation is based on array ids.
Please get in touch with @friedow to get an appropriate solution.